ui/src/routes/_layout/organizations/$orgId/products/$productId/repositories/$repoId/runs/$runIndex/index.tsx

```diff
@@ -218,18 +235,44 @@ const RunComponent = () => {
               <TableCell>Logs</TableCell>
               <TableCell>
                 <Button
-                  onClick={() => {
-                    // Get the list of steps that were run to only request logs for those steps. The `config` step is
-                    // always run implicitly and not part of the jobs, so add that manually.
-                    const jobNames = Object.keys(ortRun.jobs).concat('config');
-
-                    downloadLogs(ortRun.id, '', jobNames.join(','));
-                  }}
+                  onClick={() => handleLevelChoice(level)}
                   variant='outline'
-                  className='font-semibold text-blue-400'
+                  className={'rounded-r-none font-semibold text-blue-400'}
                 >
-                  Download log archive
+                  Download {level} log archive
                 </Button>
+                <DropdownMenu>
+                  <DropdownMenuTrigger asChild>
+                    <Button
+                      variant='outline'
+                      className={'rounded-l-none border-l-0 px-2 pt-1'}
+                    >
+                      <ChevronDownIcon />
+                    </Button>
+                  </DropdownMenuTrigger>
+                  <DropdownMenuContent>
+                    <DropdownMenuRadioGroup
+                      value={level}
+                      onValueChange={(value) => {
+                        setLevel(value);
+                        handleLevelChoice(value);
+                      }}
+                    >
+                      <DropdownMenuRadioItem value='ERROR'>
+                        Download ERROR log archive
+                      </DropdownMenuRadioItem>
+                      <DropdownMenuRadioItem value='WARN'>
+                        Download WARN log archive
+                      </DropdownMenuRadioItem>
+                      <DropdownMenuRadioItem value='INFO'>
+                        Download INFO log archive
+                      </DropdownMenuRadioItem>
+                      <DropdownMenuRadioItem value='DEBUG'>
+                        Download DEBUG log archive
+                      </DropdownMenuRadioItem>
+                    </DropdownMenuRadioGroup>
```

Comment:
As this is not really a radio, a better way could be something like:

```typescript
<DropdownMenuItem
  onClick={() => {
    downloadLogs(ortRun.id, "ERROR", jobNames.join(','));
  }
>
  Download ERROR log archive
</DropdownMenuItem>
```

Reply:
Umm what? I totally think this is a radio, as radio items indicate what value was chosen before, which I think is useful if the user wants to save logs for different levels from the same run. And isn't your example quite repetitive?

Reply:
[Radio menu](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/radio) sets a selected value, that can then be used in a form. This is just a list of buttons with differenc actions executed immediately on click.

Reply:
> [Radio menu](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/radio) sets a selected value, that can then be used in a form. This is just a list of buttons with differenc actions executed immediately on click.

I'd at least group them then with `<DropdownMenuGroup>`, to lift the state handling and function calling to the group component, instead of repeating almost exactly the same code for each item.

Reply:
I don't think there's need for state handling with `onClick`s, or what state do you need? As for repetition, it's so little and abstraction brings more complexity, so I'd just repeat it for the levels as they're so few and simple.